src/peakrdl_sv/exporter.py

```diff
@@ -109,6 +109,23 @@ def get_cpuif_bit_slice(self) -> str:
             return f"{msb}"
         else:
             return f"{msb}:{lsb}"
+        
+    def get_n_bits(self, bittype: list[str]) -> int:
```

Comment:
Looking at the way this function is used, it seems you only call it with arguments ['d', 'de'] or ['q', 'qe', 'qre']. For the first case this is equivalent to (I believe)

```
def get_write_bits(self) -> int:
	"""d and qe"""

	if not self.is_hw_writable:
		return 0

	return self.width + int(not self.external)
```

and the second case is equivalent to:
```
def get_read_bits(self) -> int:
	"""q, qe and qre""" 

	if not self.is_hw_readable:
		return 0

	return self.width + int(self.needs_qe) + int(self.needs_qre) 
```

And then we could have a function which gets all the field's bits (though currently not very useful) as:

```
def get_n_bits(self, bittype: list[str]) -> int:
    """Returns the number of bits used in this field.
    """
    
    return self.get_readable_bits() + self.get_writable_bits()
```

This approach is somewhat less flexible as you couldn't for example extract the ['qe', 'de'] bits, as you could in the current implementation, but are we ever going to want to do that? 

Within the mako template we'd then replace the call to `f.get_n_bits(['q', 'qe', 're'])` with `f.get_read_bits()`. I think this expresses intent more clearly as we 'want the read related bits' but the responsibility is on the `Field` class to decide which read related fields it actually has and tally up their lenghts.

Reply:
I think that's a fair comment and I agree with the conclusions.  I just copied the code from regtool to avoid thinking about it.  I can update it.

Reply:
Sounds good, feel free to use an alternative name if you think the functions above can be better named.